### PR TITLE
Feature/overview docs

### PR DESCRIPTION
**Pull request by talton-rmn**
Fixes "it's" -> "its" in multiple places.

Adds a better overview page for shield with gifs.

### CODE REVIEW

File: docs/start/alternatives.md [hunk 1]
```diff
@@ -10,14 +10,14 @@ Shield fits into a category of software called an API Gateway.  There are severa
...
-discover what endpoints an upstream service has and automatically build it's routing table accordingly.  In the projects
+discover what endpoints an upstream service has and automatically build its routing table accordingly.  In the projects
 below, you must explicitly configure each endpoint (if the project handles routing between services at all).
 
 <div class="panel panel-info">
   <div class="panel-heading">Correctness of Comparisons</div>
   <div class="panel-body">
     <p>
-        We try to be objective as possible in these comparisons.  However, it's possible that we missed some features in 
+        We try to be objective as possible in these comparisons.  However, its possible that we missed some features in
```
Comment (hubert) [inline docs/start/alternatives.md]: this one should be "it's"

File: docs/start/examples/listener.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ activeSubPill: listener
 # Adding Listeners
 
 In the [Stateful Middleware]({{ site.baseurl }}/start/examples/stateful_middleware) example, we set up Shield to start
-doing some heavy lifting.  However, it's hard to appreciate our work if we can't keep track of what Shield has done.
+doing some heavy lifting.  However, its hard to appreciate our work if we can't keep track of what Shield has done.
```
Comment (hubert) [inline docs/start/examples/listener.md]: it's

File: docs/start/overview.md [hunk 3]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+For those unfamiliar with the API Gateway pattern, the gateway acts as the primary entrypoint into a microservice 
+ecosystem.  Depending on how it's configured, both internal and external clients may send requests to the gateway for 
+delivery to the appropriate upstream service.
 
-Microservice Integration
+## Shield: Service Discovery 
 
-1. Service Discovery
-1. Endpoint Discovery (Swagger)
+To make the life of an infrastructure engineer as easy as possible, your tooling must be as automated as possible.
```
Comment (nickvollmar) [inline docs/start/overview.md]: Grammatically this sentence is split between the second and third person. 

I suggest reworking it to be third person, to agree with the rest of this document. Maybe something like:

* [_restructured_]: Automated tooling makes life easier for an infrastructure engineer.
* [_"funny"_] Tools are easier to work with when they do all the work themselves.

File: docs/start/overview.md [hunk 4]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+## Shield: The Router
+
+When Shield is in front of multiple unique services, it needs to know where to forward the request on to.  To
+solve this routing problem, Shield builds a routing table for all the upstream endpoints that were discovered
+during the Service Discovery phase.  The routing table then maps a request to a service by looking at the HTTP method, 
+path, and content type of the request and desired response.
+
+<img src="{{ site.baseurl }}/assets/images/routing.gif" class="img-responsive">
+
+This routing is fast, on the order of 0.01ms, even when choosing between hundreds of endpoints exposed by the upstream
```
Comment (nickvollmar) [inline docs/start/overview.md]: choosing _among_ hundreds of endpoints :D

File: docs/start/overview.md [hunk 5]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+services.  
+
+Shield is smart when multiple upstream endpoints could be used to satisfy the request, and uses the [most
+specific endpoint]({{ site.baseurl }}/start/lifecycle/endpoint_sorting) possible when deciding which upstream to proxy
+to.
+
+## Shield: The Load Balancer
+
+When Shield is in front of multiple instances of the same service, it will load balance requests across the different
```
Comment (nickvollmar) [inline docs/start/overview.md]: A reader with no context might be confused by the narrative of this paragraph, I think. In particular, "load balancing occurs on a per endpoint basis" comes across as a plot twist.

To make that clear from the beginning, we could:
1. start this paragraph mentioning that Shield load-balances on a per-endpoint basis
2. go on to describe the ramifications: you get load-balancing "for free" whether you have multiple instances of the same service, or different services advertising the same endpoint.

File: docs/start/overview.md [hunk 6]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+<img src="{{ site.baseurl }}/assets/images/load_balancing.gif" class="img-responsive">
+
+Shield has weighted load balancing logic, and will smoothly transition between weight values.  In the example above,
+"Service B" could be the next version of "Service A".  Updating the weight on Service B would smoothly transition the 
+load to the new deploy, giving it ample time to scale up if needed.  This is ideal for graceful blue/green deployments.
+
+
+## Shield: The Proxy
+
+Shield takes advantage of it's position as a proxy to help guard the upstream services during times of trouble.  Each
```
Comment (nickvollmar) [inline docs/start/overview.md]: _its_ position as a proxy

File: docs/start/overview.md [hunk 7]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+"Service B" could be the next version of "Service A".  Updating the weight on Service B would smoothly transition the 
+load to the new deploy, giving it ample time to scale up if needed.  This is ideal for graceful blue/green deployments.
+
+
+## Shield: The Proxy
+
+Shield takes advantage of it's position as a proxy to help guard the upstream services during times of trouble.  Each 
+upstream service is guarded by a [Circuit Breaker](http://doc.akka.io/docs/akka/2.4.4/common/circuitbreaker.html) so that
+Shield will automatically re-route or shed traffic if the upstream service is experiencing consistent failures or slowdowns.  
+Additionally, each endpoint on the upstream service has it's own dedicated Circuit Breaker so that one endpoint that
```
Comment (nickvollmar) [inline docs/start/overview.md]: _its_ own dedicated CB

File: docs/start/overview.md [hunk 8]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+
+## Shield: The Proxy
+
+Shield takes advantage of it's position as a proxy to help guard the upstream services during times of trouble.  Each 
+upstream service is guarded by a [Circuit Breaker](http://doc.akka.io/docs/akka/2.4.4/common/circuitbreaker.html) so that
+Shield will automatically re-route or shed traffic if the upstream service is experiencing consistent failures or slowdowns.  
+Additionally, each endpoint on the upstream service has it's own dedicated Circuit Breaker so that one endpoint that 
+relies on a flaky resource does not bring down the whole service.  
+
+In the example below, **Service B** has experienced a failure and it's circuit breaker has opened, resulting in Shield
```
Comment (nickvollmar) [inline docs/start/overview.md]: _its_ circuit breaker

File: docs/start/overview.md [hunk 9]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+immediately returning a failure response.
+
+<img src="{{ site.baseurl }}/assets/images/proxying.gif" class="img-responsive">
+
+While the lingua franca of the web is HTTP, it's very common to use a different method for communicating between the 
+services in your architecture.  Shield supports each upstream service having a different transport mechanism.  The 
+incoming request will be translated and sent across the wire in the dialect that the service supports.  Shield currently
+handles HTTP, HTTPS, and AWS Lambda, with more to come in the near future.
+
+Shield will also handle the compression of the requests and responses, inflating or deflating them based off the
```
Comment (nickvollmar) [inline docs/start/overview.md]: based on the preferences

File: docs/start/overview.md [hunk 10]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+
+Shield will also handle the compression of the requests and responses, inflating or deflating them based off the 
+preferences of the clients and the services.
+
+
+## Shield: The Middleware Engine
+
+Under the hood, Shield uses [Akka's](http://akka.io) Actor model to power request and response middleware.  These 
+middleware can perform common operations like caching, rate limiting, and authentication.  These middleware are all 
+driven by Shield configuration, including the ability to set timeouts that the middleware must respond by.  If the
```
Comment (nickvollmar) [inline docs/start/overview.md]: I'm not usually anal-retentive about dangling prepositions, but in this case the Principle of Least Surprise kicks in: "... timeouts that the middleware must respond by" doesn't feel idiomatic to me.

Maybe: "timeouts before which the middleware must respond."

File: docs/start/overview.md (hunk 7 above)
Comment (nickvollmar) [reply docs/start/overview.md]: The first "that" makes me read this sentence with an amusing intonation.

* "We all have _that one relative_ who thinks he can grill and ends up burning the hot dogs at the family reunion."
* "its own dedicated Circuit Breaker so _that one endpoint_ that relies on a flaky resource..."

I would eliminate the first "that": "so one endpoint that relies..."

File: docs/start/overview.md [hunk 11]
```diff
@@ -8,21 +8,100 @@ activePill: overview
...
+
+Listeners are similar to middleware, but they receive the request and response pair after the response has already been
+sent to the client.  This allows listeners to asynchronously do their processing without impacting the client experience.
+Common use cases here include logging and analytics.  Like their middleware brethren, listeners are controlled via 
+configuration.
+
+
+## Shield: Production Ready
+
+Shield has been powering RetailMeNot's API since 2016 and has proven to be stable and fast across billions of requests.
```
Comment (nickvollmar) [inline docs/start/overview.md]: Maybe include the month, so it's apparent that Shield's been in front of PROD for over a year :D

(Unless it hasn't been a year. In that case don't include the month, lol.)

**Review by nickvollmar — APPROVED** [review]
Great job! These are some extensive docs and motivation. And I am glad the meme diagrams are back.

I have one paragraph-structure suggestion and a few minor grammar points.

**Pull request merged**